### PR TITLE
[FIX] web,*: fix dark-mode

### PR DESCRIPTION
**Pull request by chgo-odoo**
Adaptations  for enterprise's dark-mode 

Needed by :
- https://github.com/odoo/enterprise/pull/44458
- https://github.com/odoo/o-spreadsheet/pull/3051

task-3201038

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by chgo-odoo** [comment]
@mano-odoo

**Comment by chgo-odoo** [comment]
Hi @stefanorigano 👋

These two PR's are ready for you to have a look at it, could you have a look at the changes applied on this PR and the Enterprise one?

Thanks 👍

**Review by Brieuc-brd — CHANGES_REQUESTED** [review]
Just a few comments about the code :) 

- Don’t forget to check for trailing spaces : this may help you : https://marketplace.visualstudio.com/items?itemName=shardulm94.trailing-spaces
- e2c9ca26de84d572eb20098cd2c0bb093ce12433 : "This commit~~s~~" 
- c2f2e119f7b79d44ea5b184cc2fd3c64a4583284 commit description : is the commit hash still correct ?
- Could we squash these two commits since they target the same module ? : cf9e75cae20c51fef02f46671975fcf4812c5a8d and a42bd6e8dc8993fa8f89fc0faa8765862ddb85dc

For the rest, I'll add it in your pad 🙂

**Comment by chgo-odoo** [comment]
Hello @Brieuc-brd 👋

As mentioned last week, this PR and [its Enterprise counterpart](https://github.com/odoo/enterprise/pull/44458) is ready for review.

There is still some implementation that needs to be done for : 

👉 The sign module
👉 The Chart.js lib

Could you have a look at these two PR's?

Thanks 👍

File: addons/web/static/src/core/model_field_selector/model_field_selector_popover.xml [hunk 1]
```diff
@@ -25,7 +25,7 @@
                     <div class="o_model_field_selector_popover_search mt-2">
                         <input type="text"
                                placeholder='Search...'
-                               class="o_input"
+                               class="o_input border p-1 bg-view rounded-1"
```
Comment (Brieuc-brd) [inline addons/web/static/src/core/model_field_selector/model_field_selector_popover.xml:28]: Can you reorder the classes ? :)

File: addons/web/static/src/core/model_field_selector/model_field_selector_popover.xml [hunk 2]
```diff
@@ -35,14 +35,14 @@
                 <ul class="o_model_field_selector_popover_page">
                     <t t-foreach="state.page.fieldNames" t-as="fieldName" t-key="fieldName">
                         <t t-set="fieldDef" t-value="state.page.fieldDefs[fieldName]" />
-                        <li class="o_model_field_selector_popover_item d-flex" t-att-class="{ 'active': fieldName === state.page.focusedFieldName }" t-att-data-name="fieldName">
-                            <button class="o_model_field_selector_popover_item_name btn flex-fill text-start" t-on-click="() => this.selectField(fieldDef)">
+                        <li class="o_model_field_selector_popover_item d-flex border-bottom" t-att-class="{ 'active': fieldName === state.page.focusedFieldName }" t-att-data-name="fieldName">
+                            <button class="o_model_field_selector_popover_item_name btn btn-light rounded-0 flex-fill text-start" t-on-click="() => this.selectField(fieldDef)">
```
Comment (Brieuc-brd) [inline addons/web/static/src/core/model_field_selector/model_field_selector_popover.xml:39]: `rounded-0` after `flex-fill`

File: addons/mail/static/src/discuss_app/autoresize_input.xml [hunk 3]
```diff
@@ -3,7 +3,7 @@
 
 <t t-name="mail.AutoresizeInput" owl="1">
     <input
-        class="o-mail-AutoresizeInput text-truncate px-1 border-1 border-transparent"
+        class="o-mail-AutoresizeInput o_input border-1 text-truncate px-1"
```
Comment (Brieuc-brd) [inline addons/mail/static/src/discuss_app/autoresize_input.xml:6]: px-1 before border-1

File: addons/mail/static/src/web/discuss_app/sidebar.xml [hunk 4]
```diff
@@ -44,7 +44,7 @@
             <t t-esc="mailbox.name"/>
         </div>
         <div t-attf-class="flex-grow-1 {{ mailbox.counter === 0 ? 'me-3': '' }}"/>
-        <div t-if="mailbox.counter > 0" t-attf-class="badge rounded-pill {{ mailbox.id === 'starred' ? 'bg-400 text-light' : 'text-bg-primary' }} ms-1 me-3">
+        <div t-if="mailbox.counter > 0" t-attf-class="badge rounded-pill {{ mailbox.id === 'starred' ? 'bg-400 text-bg-300' : 'text-bg-primary' }} ms-1 me-3">
```
Comment (Brieuc-brd) [inline addons/mail/static/src/web/discuss_app/sidebar.xml:47]: Are you sure to keep `bg-400` ?

**Review by Brieuc-brd — CHANGES_REQUESTED** [review]
Hey @chgo-odoo @mano-odoo 👋 
Some comments below :) 
Also, you should double check your commit titles & descriptions, I've seen a few mistakes 👀

**Review by stefanorigano — APPROVED** [review]
LGTM, thanks :+1

**Review by stefanorigano — APPROVED** [review]
LGTM, thanks :+1

File: addons/spreadsheet/static/src/o_spreadsheet/o_spreadsheet.js [hunk 5]
```diff
@@ -43178,6 +43178,7 @@
     css /* scss */ `
   .o-border-selector {
     padding: 4px;
+    background-color: white;
```
Comment (LucasLefevre) [inline addons/spreadsheet/static/src/o_spreadsheet/o_spreadsheet.js:43181]: Hello :)
Thanks for the fixes
`o_spreadsheet.js` and `o_spreadsheet.xml` are automatically generated by the [o-spreadsheet](https://github.com/odoo/o-spreadsheet) lib build tools. Your changes would be overwritten the next time we update the lib (once every 1-2 weeks).
These changes need to be in o-spreadsheet repo.
Do you want to do it (we don't have a -dev repo, you can push on origin ;)) ?
Or do you prefer to let us do it ?

Comment (stefanorigano) [reply addons/spreadsheet/static/src/o_spreadsheet/o_spreadsheet.js:43181]: Hello @LucasLefevre , thanks for the heads-up  👍 
Changes has been applied  here https://github.com/odoo/o-spreadsheet/pull/3051

File: addons/spreadsheet/static/src/o_spreadsheet/o_spreadsheet.js
Comment (pparidans) [inline addons/spreadsheet/static/src/o_spreadsheet/o_spreadsheet.js:1]: This file is generated (spreadsheet build) ; it shouldn't be edited.

**Review by pparidans — CHANGES_REQUESTED** [review]
Looks good.

Small change to do related to the spreadsheet build.

As discussed, even if merged in stable release and updating a huge chunk of templates, applying only the CSS changes looks OK. So LGTM.

File: addons/spreadsheet/static/src/o_spreadsheet/o_spreadsheet.js
Comment (chgo-odoo) [reply addons/spreadsheet/static/src/o_spreadsheet/o_spreadsheet.js:1]: Thanks @pparidans 👋

As SRI addressed the spreadsheet changes in another PR, we will simply remove our changes from this PR. MANO is doing it right now

cc @stefanorigano

**Comment by pparidans** [comment]
@robodoo r+

**Pull request closed without merging**